src/predictor/gpu_predictor.cu

```diff
@@ -225,13 +225,14 @@ class GPUPredictor : public xgboost::Predictor {
     auto& offsets = *out_offsets;
     offsets.resize(devices_.Size() + 1);
     offsets[0] = 0;
+    const size_t last = data.ConstHostVector().back();
```

Comment:
This may still cause unnecessary creation of a CPU copy if the data is originally present only on the GPU.

On the other hand, this value is equal (at least in the single-batch case) to `batch.data.Size()`, as it is the total number of entries. Could you use that value instead?

Reply:
Done.